internal/app/singularity/registry.go

```diff
@@ -0,0 +1,127 @@
+package singularity
+
+import (
+	"context"
+	"fmt"
+
+	scs "github.com/sylabs/scs-library-client/client"
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/library"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+	"github.com/sylabs/singularity/internal/pkg/util/uri"
+	"github.com/sylabs/singularity/pkg/signing"
+)
+
+var errNotInCache = fmt.Errorf("image was not found in cache")
+
+// Library is a Registry implementation for Sylabs Cloud Library.
+type Library struct {
+	keystoreURI string
+
+	client *scs.Client
+	cache  *cache.Handle
+}
+
+// NewLibrary initializes and returns new Library ready to  be used.
+func NewLibrary(client *scs.Client, cache *cache.Handle, keystoreURI string) *Library {
```

Comment:
Would it be better to have This accept a `client.Config{}` and call `client.NewClient()` here so that we don't need two steps every time we want a new library object?

Reply:
Well, on the one hand yes, that will save lines.
On the other hand ideally an object accepts dependencies rather than creates them itself, that was the initial motivation.

But I see that `NewClient` simply records fields from config, so it may be reasonable to update.
Unless we may want to bypass `NewClient` in tests (that are not in this PR, but will be in place soon).

Reply:
I'm kinda in agreement with @ikaneshiro here for the reason that both...

- a `Library` is useless without a client to access it
- a `Client` is not to be used outside of the context of a Library

...seem true?

Then the internal state of a `Library` holding a `Client` is preferable to having to write it out longhand every time?

Reply:
@dctrud You convinced me, arguments seem true.